engine/runtime-compiler/src/main/java/org/enso/compiler/phase/ImportResolverAlgorithm.java

```diff
@@ -166,6 +165,96 @@ private ResolvedType tryResolveAsTypeNew(Import name) {
     return type.orElse(null);
   }
 
+  private ResolvedConstructor tryResolveAsConstructor(Import imp) {
+    var parts = partsForImport(imp);
+    if (parts.size() < 3) {
+      return null;
+    }
+    var constructors = definedConstructors(imp);
```

Comment:
Would it make sense to return a `Map` here? Then the O(N) stream filter could potentially become an O(log N) / O(1) map access.

Reply:
Also looking at the code it seems like we are constructing this on each call. Have you considered caching the result? I'm not sure how many times this is called, but I'd assume it can be called a lot on heavily used modules, so avoiding to re-construct this List (or Map) every time could give us some additional speedup.